api/include/opentelemetry/trace/trace_state.h

```diff
@@ -0,0 +1,168 @@
+#include <vector>
+#include "opentelemetry/nostd/string_view.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace trace {
+public abstract class TraceState {
+  private:
+    static const int KEY_MAX_SIZE = 256;
+    static const int VALUE_MAX_SIZE = 256;
+    static const int MAX_KEY_VALUE_PAIRS = 32;
+    static const TraceState DEFAULT = TraceState.builder().build();
+
+    static TraceState create(std::vector<Entry> entries) {
+      // I don't know what exactly this is and how to replace this
+      return new AutoValue_TraceState(Collections.unmodifiableList(entries));
+    }
+
+    // Value is opaque string up to 256 characters printable ASCII RFC0020 characters (i.e., the range
+    // 0x20 to 0x7E) except comma , and =.
+    static bool validateValue(nostd::string_view value) {
+      if (value.length() > VALUE_MAX_SIZE || value[value.length() - 1] == ' ' /* '\u0020' */) {
+        return false;
+      }
+      for (int i = 0; i < value.length(); i++) {
+        char c = value[i];
+        if (c == ',' || c == '=' || c < ' ' /* '\u0020' */ || c > '~' /* '\u007E' */) {
+          return false;
+        }
+      }
+      return true;
+    }
+
+    static bool isNumberOrDigit(char ch) {
+      return (ch >= 'a' && ch <= 'z') || (ch >= '0' && ch <= '9');
+    }
+
+    // Key is opaque string up to 256 characters printable. It MUST begin with a lowercase letter, and
+    // can only contain lowercase letters a-z, digits 0-9, underscores _, dashes -, asterisks *, and
+    // forward slashes /.  For multi-tenant vendor scenarios, an at sign (@) can be used to prefix the
+    // vendor name.
+    // todo: benchmark this implementation
+    static bool validateKey(nostd::string_view key) {
+      if (key.length() > KEY_MAX_SIZE || key.isEmpty() || !isNumberOrDigit(key[0]))) {
+        return false;
+      }
+      int atSeenCount = 0;
+      for (int i = 1; i < key.length(); i++) {
+        char c = key[i];
+        if (!isNumberOrDigit(c) && c != '_' && c != '-' && c != '@' && c != '*' && c != '/') {
+          return false;
+        }
+        if ((c == '@') && (++atSeenCount > 1)) {
+          return false;
+        }
+      }
+      return true;
+    }
+
+  public:
+    // Builder class for TraceState.
+    static const class Builder {
+      private:
+        const TraceState parent;
+        std::vector<Entry> entries;
```

Comment:
You can't use vector in API headers.

Reply:
So are there any work-around of this as this data-structure seems important to keep?